### PR TITLE
Fix new-lines rule on Python 3

### PR DESCRIPTION
**Pull request by bz2**
The fix is reasonably simple, to add the test that fails on Python 3 I also did some refactoring of `test_cli.py` setup. See the individual commits for full details.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -377,63 +350,46 @@ def test_run_non_ascii_file(self):
             locale.setlocale(locale.LC_ALL, 'C.UTF-8')
         except locale.Error:
             locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
+        self.addCleanup(locale.setlocale, locale.LC_ALL, loc)
```
Comment (adrienverge) [inline tests/test_cli.py]: Can you revert this?
For correct testing and for reproductibility, it's better if the original locale is restored *at the end of this test*, not *at the end of all tests*.

Comment (bz2) [reply tests/test_cli.py]: [`addCleanup()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) happens after `tearDown()` - at the end of every test, not at the end of all.

**Comment by bz2** [comment]
Thanks for the prompt review! I'll rearrange the commits as requested.

**Comment by bz2** [comment]
@adrienverge Have rearranged and squashed to two commits as requested.

File: tests/test_cli.py (hunk 1 above)
Comment (adrienverge) [reply tests/test_cli.py]: OK, thanks for the clarification!

**Pull request merged**